lms/djangoapps/experiments/tests/test_views_custom.py

```diff
@@ -3,30 +3,99 @@
 """
 from __future__ import absolute_import
 
+from uuid import uuid4
+import six
+
 from django.urls import reverse
 from rest_framework.test import APITestCase
 
+from course_modes.tests.factories import CourseModeFactory
 from lms.djangoapps.course_blocks.transformers.tests.helpers import ModuleStoreTestCase
 from student.tests.factories import UserFactory
 
+from openedx.core.djangoapps.waffle_utils.testutils import override_waffle_flag
+from xmodule.modulestore.tests.factories import CourseFactory
+
+from lms.djangoapps.experiments.views_custom import MOBILE_UPSELL_FLAG
+
 CROSS_DOMAIN_REFERER = 'https://ecommerce.edx.org'
 
 
-class Rev934Tests(APITestCase, ModuleStoreTestCase):
-    def test_logged_in(self):
-        """Test mobile app upsell API"""
+class Rev934LoggedOutTests(APITestCase):
+    def test_not_logged_in(self):
+        """Test mobile app upsell API is not available if not logged in"""
         url = reverse('api_experiments:rev_934')
-        user = UserFactory()
 
         # Not-logged-in returns 401
         response = self.client.get(url)
         self.assertEqual(response.status_code, 401)
 
-        # No-course-id returns show_upsell false
+
+class Rev934Tests(APITestCase, ModuleStoreTestCase):
+    """Test mobile app upsell API"""
+    @classmethod
+    def setUpClass(cls):
+        super(Rev934Tests, cls).setUpClass()
+        cls.url = reverse('api_experiments:rev_934')
+
+    def setUp(self):
+        super(Rev934Tests, self).setUp()
+        user = UserFactory(username='robot-mue-1-6pnjv')  # robot-mue-0-l23fi
```

Comment:
Why is this the username? And why do you have a comment with a different username?

Reply:
This username buckets into the experiment; the comment is extraneous, I'll remove it in the next round